### PR TITLE
Fix for "pegasus to avro translation of UnionWithAlias RecordFields does not have field properties". 

### PR DESCRIPTION
**Pull request by junchuanwang**
Add more tests for UnionWithAlias translation

### CODE REVIEW

File: data-avro/src/test/java/com/linkedin/data/avro/TestSchemaTranslator.java [hunk 1]
```diff
@@ -75,6 +75,162 @@ public void testTranslateDefaultBackwardsCompatibility()
...
+                    "      ]," +
+                    "      \"compliance\": {" +
+                    "        \"/fieldDiscriminator\": \"NONE\"," +
+                    "        \"/string\": \"NONE\"" +
+                    "      }" +
+                    "    }" +
+                    "  ]" +
+                    "}"
+            },
+            // Test : field properties overrides Typeref properties
```
Comment (karthikbalasub) [inline data-avro/src/test/java/com/linkedin/data/avro/TestSchemaTranslator.java:165]: You should add a test for the case where the field specifies properties for a union member and another where it specifies property for "fieldDiscriminator" field. The test should check that the properties are propagated correctly to the nested fields representing the members.

Comment (karthikbalasub) [reply data-avro/src/test/java/com/linkedin/data/avro/TestSchemaTranslator.java:165]: This should be for union with alias.

Comment (junchuanwang) [reply data-avro/src/test/java/com/linkedin/data/avro/TestSchemaTranslator.java:165]: @karthikbalasub I don't think we are propagating the `"on-the-field" union member annotation` to `nested fields` in the translator. Isn't that the job of annotation processor?

Comment (junchuanwang) [reply data-avro/src/test/java/com/linkedin/data/avro/TestSchemaTranslator.java:165]: @karthikbalasub 
> This should be for union with alias.
You are right

File: data-avro/src/main/java/com/linkedin/data/avro/SchemaToAvroJsonEncoder.java [hunk 2]
```diff
@@ -417,30 +425,27 @@ protected void encodeFieldProperties(RecordDataSchema.Field field) throws IOExce
...
+    BinaryOperator<Object> propertyMergeLogic = (originalPropertyContent, inheritedPropertyContent) ->
+    {
       if (originalPropertyContent instanceof Map && inheritedPropertyContent instanceof Map)
       {
         Map<String, Object> mergedMap = new DataMap((Map<String, Object>) originalPropertyContent);
         ((Map<String, Object>) inheritedPropertyContent).forEach(mergedMap::putIfAbsent);
         return mergedMap;
-      }
-      else
+      } else
```
Comment (nickibi) [inline data-avro/src/main/java/com/linkedin/data/avro/SchemaToAvroJsonEncoder.java:441]: nit: unnecessary format change

**Review by nickibi — APPROVED** [review]
minor nit, otherwise LGTM

**Review by karthikbalasub — APPROVED** [review]

**Review by karthikbalasub — APPROVED** [review]

**Pull request merged**